language/documentation/book/src/unit-testing.md

```diff
@@ -0,0 +1,332 @@
+# Unit Tests
+
+Unit testing for Move adds three new annotations to the Move source language:
+
+* `#[test]`
+* `#[test_only]`, and
+* `#[expected_failure]`.
+
+They respectively mark a function as a test, mark a module or module member (`use`, function, or struct) as code to be included for testing only, and mark that a test is expected to fail/abort. These annotations can be placed on a function with any visibility. Whenever any module or module member is annotated as `#[test_only]` or `#[test]`, it will not be included in the compiled bytecode unless it is compiled for testing.
+
+## Testing Annotations: Their Meaning and Usage
+
+Both the `#[test]` and `#[expected_failure]` annotations can be used either with or without arguments.
+
+Without arguments, the `#[test]` annotation can only be placed on a function with no parameters. This annotation simply marks this function as a test to be run by the unit testing harness.
+
+```
+#[test] // OK
+fun this_is_a_test() { ... }
+
+#[test] // Will fail to compile since the test takes an argument
+fun this_is_not_correct(arg: signer) { ... }
+```
+
+A test can also be annotated as an `#[expected_failure]`. This annotation marks that the annotated test should abort — however the exact abort code is not checked. The test will be marked as failing only if the test fails with a non-abort error, or does not abort. Only functions that have the `#[test]` annotation can also be annotated as an  #`[expected_failure]`.
```

Comment:
Could you provide an example, either here or in a subsequent paragraph or section, about what a "non-abort error" means? Is a bytecode verification or VM error considered a non-abort error?

If my understanding is correct, then I think I'd find this clearer as:

```suggestion
A test can also be annotated as an `#[expected_failure]`. This annotation marks that the annotated test should abort — however the exact abort code is not checked. The test will pass only if an `abort` statement is executed that halts the test's execution. Only functions that have the `#[test]` annotation can also be annotated as an  #`[expected_failure]`.
```

Reply:
Ah, so this also covers other errors such as if you do a `borrow_global` on a resource that doesn't exist. I'll update with an example